lib/shared.bash

```diff
@@ -11,25 +11,32 @@ esac
 
 vault_auth() {
   local server="$1"
-  local auth_params=''
-  # role is defined in Vault Configuration
-  # -header=<foo> # used to set X-Auth
-  # BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD - aws
-  # BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_HEADER
-  # BUILDKITE_PLUGIN_VAULT_SECRETS_ROLE
-  [ -n "${server:-}" ] && auth_params="${auth_params} -address=${server}"
-  [ -n "${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD:-}" ] && auth_params="${auth_params} -method=${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD}"
-  [ -n "${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_HEADER:-}" ] && auth_params="${auth_params} -header_value=${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_HEADER}"
-  [ -n "${BUILDKITE_PLUGIN_VAULT_SECRETS_ROLE:-}" ] && auth_params="${auth_params} role=${BUILDKITE_PLUGIN_VAULT_SECRETS_ROLE}"
-
-  if [ -n "${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD:-}" ] ; then
-    # don't output the token to log, even though it's a temporary token
-    # shellcheck disable=SC2086
-    vault auth $auth_params | grep -v ^token:
+
+  # Currently we only support AppRole authentication.
+  # These values are referenced when authenticating to the Vault server:
+  #   BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD - approle
+
+  #   RoleID and SecretID should be stored securely on the agent, there are probably better ways to do this, but here is a start
+  #   We'll use these two values for the RoleID and SecretID:
+  #     BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_SECRET_ID
+  #     BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_ROLE_ID
+  #
+  #   For now, you will need to define the secret ID oustide of the plugin, though this will probably change.
+
+   # approle authentication
+  if [ "${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD:-}" = "approle" ]; then
+    [ -n "${BUILDKITE_PLUGIN_VAULT_SECRETS_ROLE_ID:-}" ]
+    [ -n "${BUILDKITE_PLUGIN_VAULT_SECRETS_SECRET_ID:-}" ]
```

Comment:
what are these two lines supposed to do? If you want to fail the execution (using the `-e` option set somewhere else I believe you should use `${VARIABLE:?Error to display when variable is empty or not set}`

Reply:
Good catch - they don't do anything. Removed ✅